src/main/java/com/epam/training/gen/ai/controller/ChatController.java

```diff
@@ -20,15 +27,28 @@ public class ChatController {
 
   private final ChatCompletionService chatCompletionService;
   private final Kernel kernel;
-  private final InvocationContext invocationContext;
+  private final ChatHistory chatHistory;
+  private final EpamDialClient dialClient;
 
   @PostMapping
   public ChatOutputDto chat(@RequestBody ChatInputDto input) {
```

Comment:
I would recommend using ChatPromptDto and ChatCompletionDto to align with standard terminology.
Have you considered extending ChatInputDto to include LM settings such as temperature, topP, and init invocation context for each call based on passed params? 
This would allow you to gather results for the final part of the task, 'Comparison of Results.'

Reply:
What standard terminology? There is no standard defined in the task.
The goal of the task is to learn about AI - not DTOs, let's not get sidetracked.

As for temperature configuration - the task #2 specifically requires it to be in config, and task #3 is based on #2

Reply:
naming is just a suggestion. not going to argue with you.
below is the list of requirements for the task. Part 3 is not done, that is why 95 instead of 100

### Evaluation Criteria

1. Configuration for Different Models

* Correctly modified the application.properties file to change the deployment name for different models (15%)
* Successfully retrieved deployment names using the provided API (15%)

2. Implementation of Model Switching

* Correctly implemented the logic to switch between different models based on the deployment name (15%)
* Ensured that the application can call different models from the Dial service (15%)

3. Comparison of Results

* Compared results for the same prompts using different models (10%)
* Compared results using different PromptExecutionSettings (10%)
* Documented observations and differences in the results (10%)

4. Functionality Testing

* Application runs without errors (5%)
* Responses are generated correctly and are relevant to the prompts (5%)